homeassistant/components/zha/siren.py

```diff
@@ -64,115 +50,50 @@ async def async_setup_entry(
         hass,
         SIGNAL_ADD_ENTITIES,
         functools.partial(
-            discovery.async_add_entities,
-            async_add_entities,
-            entities_to_create,
+            zha_async_add_entities, async_add_entities, ZHASiren, entities_to_create
         ),
     )
     config_entry.async_on_unload(unsub)
 
 
-@MULTI_MATCH(cluster_handler_names=CLUSTER_HANDLER_IAS_WD)
-class ZHASiren(ZhaEntity, SirenEntity):
+class ZHASiren(ZHAEntity, SirenEntity):
     """Representation of a ZHA siren."""
 
-    _attr_name: str = "Siren"
-
-    def __init__(
-        self,
-        unique_id: str,
-        zha_device: ZHADevice,
-        cluster_handlers: list[ClusterHandler],
-        **kwargs,
-    ) -> None:
-        """Init this siren."""
-        self._attr_supported_features = (
-            SirenEntityFeature.TURN_ON
-            | SirenEntityFeature.TURN_OFF
-            | SirenEntityFeature.DURATION
-            | SirenEntityFeature.VOLUME_SET
-            | SirenEntityFeature.TONES
-        )
-        self._attr_available_tones: list[int | str] | dict[int, str] | None = {
-            WARNING_DEVICE_MODE_BURGLAR: "Burglar",
-            WARNING_DEVICE_MODE_FIRE: "Fire",
-            WARNING_DEVICE_MODE_EMERGENCY: "Emergency",
-            WARNING_DEVICE_MODE_POLICE_PANIC: "Police Panic",
-            WARNING_DEVICE_MODE_FIRE_PANIC: "Fire Panic",
-            WARNING_DEVICE_MODE_EMERGENCY_PANIC: "Emergency Panic",
-        }
-        super().__init__(unique_id, zha_device, cluster_handlers, **kwargs)
-        self._cluster_handler: IasWdClusterHandler = cast(
-            IasWdClusterHandler, cluster_handlers[0]
+    _attr_available_tones: list[int | str] | dict[int, str] | None = {
+        WARNING_DEVICE_MODE_BURGLAR: "Burglar",
+        WARNING_DEVICE_MODE_FIRE: "Fire",
+        WARNING_DEVICE_MODE_EMERGENCY: "Emergency",
+        WARNING_DEVICE_MODE_POLICE_PANIC: "Police Panic",
+        WARNING_DEVICE_MODE_FIRE_PANIC: "Fire Panic",
+        WARNING_DEVICE_MODE_EMERGENCY_PANIC: "Emergency Panic",
+    }
+
+    def __init__(self, entity_data: EntityData, **kwargs: Any) -> None:
+        """Initialize the ZHA siren."""
+        super().__init__(entity_data, **kwargs)
+        self._attr_supported_features = SirenEntityFeature(
+            self.entity_data.entity._attr_supported_features.value  # noqa: SLF001
         )
-        self._attr_is_on: bool = False
-        self._off_listener: Callable[[], None] | None = None
 
+    @property
+    def is_on(self) -> bool:
+        """Return True if entity is on."""
+        return self.entity_data.entity.state[
+            "state"
+        ]  # TODO are we going to use info objects / state dicts or platform entity props? pylint: disable=fixme
```

Comment:
We should not add TODOs in our codebase. This pylint check is here for a reason. Please remove it and consider creating an issue to track this instead.

Reply:
Good point. I think we could/should have an `is_on` property defined for sirens in ZHA like we do for switch, light, etc.
Then, it could be accessed via `self.entity_data.entity.is_on` here.

Reply:
done (thx @TheJulianJES for the lib change)